### DISCUSSION TITLE
Search Field is Empty

### DISCUSSION THREAD

**Discussion by bhsmither** [dnGrep/dnGrep#497] [Q&A] [answered]
But why look through the file?

Maybe I am misreading the Wireshark report.

The report suggests that directories are being searched (retrieved?) that do not match the `Paths to match:` field.

Search in Folder: `\\SERVER1\inetpub`
Paths to match (Regex): `\\CubeCart6.*\\admin.*\\skins\\default\\js\\plugins\\jquery\.filetree\..*\.js.*`

For example, Wireshark says this was pulled in:
`Read Request Len:1002 Off:0 File: CubeCart6.2.4\modules\shipping\By_Percent\shipping.class.php`
Why? The /modules/ folder does not start with 'admin'.

**Comment by doug24** [comment, marked as answer]
That is correct, the file search does not use the regex to limit recursion into directories.  It enumerates all the directories and applies the regex as a filter for individual files.  Matching a regex to a file path is just a binary operation - it matches or it doesn't match the entire path.  When looking at that regex, you are parsing it in your mind and applying just the first part of it.

**Reply by doug24** [reply]
Let me try this again...
dnGrep is searching through directories under \\SERVER1\inetpub.  It gets to the directory:
`\\SERVER1\inetpub\\CubeCart6.2.4\admin\`
Should it search in this directory?  Given just an include regex, there are only two options:  1) continue searching if it matches the include regex, or 2) don't apply the include regex to directories continue into the directory and apply the include regex only to files.  If dnGrep used option 1, it the include regex does not match the directory name, so it would stop searching and move on - actually it wouldn't even get that far - that include regex will never match any directory name, so dnGrep wouldn't search anything.  So instead dnGrep does not apply the include regex to directory names, and only to full file paths.  Same goes for asterisk include patterns, they are only applied to files.

If you want to exclude everything under \modules folder, you can add an exclude pattern.  It will work with both asterisk patterns and regex patterns, but in the current implementation, only the asterisk pattern will stop recursion into a directory tree.  When using the regex pattern, the files are still enumerated and then filtered in the process of applying any other file filters.

In addition to the asterisk exclude pattern, the other filters that limit recursion into directories are the Include hidden folders filter, the Include subfolders/Max depth filter, and the Use .gitignore filter.

You can also specify multiple root folders to search in the `Folder:` input field, separating the root folders with commas or semi colons.  This way you can selectively search just some sub directories of your original search directory.

I'll note that dnGrep is not searching within your php file.  The regex file filter is applied, and the php file will be excluded from the set of files that are opened for searching.
